src/hooks/useOrders.ts

```diff
@@ -37,7 +37,8 @@ export function useOrders(): AuctionElement[] {
         .getOrders(userAddress)
         .then(filterDeletedOrders)
         .then(setOrders)
-  }, [setOrders, userAddress])
+    // updating list of orders on every block by listening on `blockNumber`
+  }, [setOrders, userAddress, blockNumber])
```

Comment:
Really like this kind of responsibility delegation, form WalletInfo to effect deps to exchangeApi

Reply:
Here is a thought. Wouldn't this approach kill the battery of a phone if we get orders on every block, decode them etc?

For stuff with events, shouldn't be more like getPastevents, from last queried block until this `blockNumber`. Possibly this just means, that it would be a performance improvement scoped INSIDE the Api implementation and the hook don't care about that --> interesting for cache logic

cc/ @Velenir @alfetopito

Reply:
From the battery perspective, will the page keep updating if not on focus? Don't know how mobile handles this, but my expectation is that whatever background task a page runs only does so when open.

For the second part, I don't really get what you mean.

Reply:
I don't think there's a difference client-side between calling `getPastEvents` and `getOrders`. Both those calls are encoded into rpc calls and sent to infura or wherever we are connected.

The only palpable difference as I see it, can be between calls and WebSocket subscriptions to events. And that is not something we can count on. Some providers do subscriptions through websockets, some do polling. MMask does polling only, as far as I can tell.

Reply:
My point was more like, your we getOrders on every block? it seams like a lot of work and I see nodes banning us.

WebSocket would solve it. 

My point is, only if there's new events for an user we have to load ALL his orders. If I know the block of the last event for the user, and I know the last block I queried the orders in the contract, I know if I have to do the call.

The battery thing, was just to point out that the app is using a lot the network, and lot of JS is being executed every few seconds, when there's no really new orders coming in for the user.